java.lang.IllegalAccessError between realm and java 16

This seem to be Java related. 

There is this in `app/build.gradle`

    sourceCompatibility JavaVersion.VERSION_1_8

However, building with Java 16 does not produce any message such as incompatible JDK, but gives a very specific error.

1. Is this error caused by JDK compatibility?
2. If so, what versions are allowed?
3. Can we customise gradle to produce useful messages when incompatible jdk is used?

```
$ gradle build

Welcome to Gradle 7.3!

Here are the highlights of this release:
 - Easily declare new test suites in Java projects
 - Support for Java 17
 - Support for Scala 3

For more details see https://docs.gradle.org/7.3/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)
Checking the license for package Android SDK Build-Tools 32 in /opt/android/licenses
License for package Android SDK Build-Tools 32 accepted.
Preparing "Install Android SDK Build-Tools 32 (revision: 32.0.0)".
"Install Android SDK Build-Tools 32 (revision: 32.0.0)" ready.
Installing Android SDK Build-Tools 32 in /opt/android/build-tools/32.0.0
"Install Android SDK Build-Tools 32 (revision: 32.0.0)" complete.
"Install Android SDK Build-Tools 32 (revision: 32.0.0)" finished.
Checking the license for package Android SDK Platform 32 in /opt/android/licenses
License for package Android SDK Platform 32 accepted.
Preparing "Install Android SDK Platform 32 (revision: 1)".
"Install Android SDK Platform 32 (revision: 1)" ready.
Installing Android SDK Platform 32 in /opt/android/platforms/android-32
"Install Android SDK Platform 32 (revision: 1)" complete.
"Install Android SDK Platform 32 (revision: 1)" finished.

> Task :lib:compileJava
Note: /home/a/IdeaProjects/alpha-wallet-android/lib/src/main/java/com/alphawallet/token/tools/XMLDSigVerifier.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/a/IdeaProjects/alpha-wallet-android/lib/src/main/java/com/alphawallet/token/tools/XMLDSigVerifier.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :app:mergeAnalyticsDebugResources
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-zh/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:204:4: Multiple substitutions specified in non-positional format of string resource string/eth_per_ticket_w_value. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:536:4: Multiple substitutions specified in non-positional format of string resource string/link_valid_until. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:702:4: Multiple substitutions specified in non-positional format of string resource string/set_price_subtext_abr. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:480:4: Multiple substitutions specified in non-positional format of string resource string/tickets_selected. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:232:4: Multiple substitutions specified in non-positional format of string resource string/total_cost. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?

> Task :app:processAnalyticsDebugMainManifest
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:name was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:theme was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:allowBackup was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present

> Task :app:mergeAnalyticsReleaseResources
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:204:4: Multiple substitutions specified in non-positional format of string resource string/eth_per_ticket_w_value. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:536:4: Multiple substitutions specified in non-positional format of string resource string/link_valid_until. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:702:4: Multiple substitutions specified in non-positional format of string resource string/set_price_subtext_abr. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:480:4: Multiple substitutions specified in non-positional format of string resource string/tickets_selected. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:232:4: Multiple substitutions specified in non-positional format of string resource string/total_cost. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-zh/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?

> Task :app:processAnalyticsReleaseMainManifest
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:name was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:theme was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:allowBackup was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present

> Task :app:processNoAnalyticsDebugMainManifest
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:name was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:theme was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/debug/AndroidManifest.xml:20:5-37:19 Warning:
        application@android:allowBackup was tagged at AndroidManifest.xml:20 to replace other declarations but no other declaration present

> Task :app:mergeNoAnalyticsDebugResources
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:204:4: Multiple substitutions specified in non-positional format of string resource string/eth_per_ticket_w_value. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:536:4: Multiple substitutions specified in non-positional format of string resource string/link_valid_until. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:702:4: Multiple substitutions specified in non-positional format of string resource string/set_price_subtext_abr. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:480:4: Multiple substitutions specified in non-positional format of string resource string/tickets_selected. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-zh/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:232:4: Multiple substitutions specified in non-positional format of string resource string/total_cost. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?

> Task :app:externalNativeBuildNoAnalyticsDebug
Execution optimizations have been disabled for task ':app:externalNativeBuildNoAnalyticsDebug' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/a/IdeaProjects/alpha-wallet-android/app/build/intermediates/cxx/Debug/1c15q2i4/obj'. Reason: Task ':app:mergeAnalyticsDebugNativeLibs' uses this output of task ':app:externalNativeBuildNoAnalyticsDebug' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :app:mergeNoAnalyticsDebugNativeLibs
Execution optimizations have been disabled for task ':app:mergeNoAnalyticsDebugNativeLibs' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/a/IdeaProjects/alpha-wallet-android/app/build/intermediates/cxx/Debug/1c15q2i4/obj'. Reason: Task ':app:mergeNoAnalyticsDebugNativeLibs' uses this output of task ':app:externalNativeBuildAnalyticsDebug' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :app:processNoAnalyticsReleaseMainManifest
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:name was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:theme was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/AndroidManifest.xml:22:5-310:19 Warning:
        application@android:allowBackup was tagged at AndroidManifest.xml:22 to replace other declarations but no other declaration present

> Task :app:externalNativeBuildNoAnalyticsRelease
Execution optimizations have been disabled for task ':app:externalNativeBuildNoAnalyticsRelease' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/a/IdeaProjects/alpha-wallet-android/app/build/intermediates/cxx/RelWithDebInfo/a5q3v2i3/obj'. Reason: Task ':app:mergeAnalyticsReleaseNativeLibs' uses this output of task ':app:externalNativeBuildNoAnalyticsRelease' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :app:mergeNoAnalyticsReleaseNativeLibs
Execution optimizations have been disabled for task ':app:mergeNoAnalyticsReleaseNativeLibs' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/a/IdeaProjects/alpha-wallet-android/app/build/intermediates/cxx/RelWithDebInfo/a5q3v2i3/obj'. Reason: Task ':app:mergeNoAnalyticsReleaseNativeLibs' uses this output of task ':app:externalNativeBuildAnalyticsRelease' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :app:mergeNoAnalyticsReleaseResources
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-zh/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:204:4: Multiple substitutions specified in non-positional format of string resource string/eth_per_ticket_w_value. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:536:4: Multiple substitutions specified in non-positional format of string resource string/link_valid_until. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:702:4: Multiple substitutions specified in non-positional format of string resource string/set_price_subtext_abr. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:480:4: Multiple substitutions specified in non-positional format of string resource string/tickets_selected. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:232:4: Multiple substitutions specified in non-positional format of string resource string/total_cost. Did you mean to add the formatted="false" attribute?
/home/a/IdeaProjects/alpha-wallet-android/app/src/main/res/values-es/strings.xml:185:4: Multiple substitutions specified in non-positional format of string resource string/total_cost_for_x_tickets. Did you mean to add the formatted="false" attribute?

> Task :app:compileAnalyticsReleaseKotlin
w: /home/a/IdeaProjects/alpha-wallet-android/app/src/main/java/com/alphawallet/app/walletconnect/WCClient.kt: (355, 9): Non exhaustive 'when' statements on enum will be prohibited in 1.7, add 'null' branch or 'else' branch instead

> Task :app:compileAnalyticsReleaseJavaWithJavac FAILED
Note: Version 10.11.0 of Realm is now available: https://static.realm.io/downloads/java/latest
Note: /home/a/IdeaProjects/alpha-wallet-android/app/src/main/java/com/alphawallet/app/ui/RedeemSignatureDisplayActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileAnalyticsReleaseJavaWithJavac'.
> java.lang.IllegalAccessError: class io.realm.processor.Utils (in unnamed module @0x55790929) cannot access class com.sun.tools.javac.code.Symbol$ClassSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x55790929

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 4 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD FAILED in 2m 48s
106 actionable tasks: 106 executed
```

--- a/BUILD.md
+++ b/BUILD.md
@@ -1,0 +1,36 @@
+# Build instruction
+
+To build everything:
+
+```
+$ ./gradle build
+```
+
+You can also use your locally installed gradle, provide that the version is higher than 7.0.2
+
+```
+$ gradle build
+```
+
+## Build dmz server
+
+DMZ server is a SpringBoot based server application that is deployed on a DMZ host, intentionally read only, stateless, have no database access and have no keys to steal from. It was made in 2017 to provide ad-hoc ad-hoc services, display token status by reading from blockchain using client wallet and provide server verification strings. There are plans to phase it out completely.
+
+To build the dmz kit:
+
+```
+$ cd dmz
+dmz $ gradle build bootJar
+
+> Task :dmz:compileJava
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+
+BUILD SUCCESSFUL in 1m 1s
+5 actionable tasks: 3 executed, 2 up-to-date
+```
+
+This will give you `build/libs/dmz.jar`
+
+Then, you can copy the jar file to the server in which this was deployed and restart the SpringBoot service.
+

--- a/BUILD.md
+++ b/BUILD.md
@@ -1,4 +1,7 @@
 # Build instruction
+### NOTE: Use JDK11
+- Android Gradle Plugin requires Java 11
+- Java 16 is not supported due to IllegalAccessError compile error when Realm tries to access the com.sun.tools.javac package
 
 To build everything:
 

--- a/README.md
+++ b/README.md
@@ -62,7 +62,7 @@ We want to give businesses the whitelabel tools they need to develop their ether
 1. [Download](https://developer.android.com/studio/) Android Studio.
 2. Clone this repository
 3. Obtain a free Infura API key and replace the one in build.gradle
-4. Build the project in AndroidStudio or Run `./gradlew build` to install tools and dependencies.
+4. Build the project in AndroidStudio or Run `./gradlew build` to install tools and dependencies. See [BUILD.md](BUILD.md) for more details.
 
 You can also build it from the commandline just like other Android apps. Note that JDK 8 and 11 are the versions supported by Android.
 

